### PR TITLE
feat: epoll based timeout & cpu time

### PR DESCRIPTION
**Pull request by Ashwin125**
- Epoll based process monitoring
- CPU time based process timeout

Co-authored-by: vigneshd332@gmail.com <vigneshd332@gmail.com>
Co-authored-by: Ram Ganesh K. R.  <Ram-20062003@users.noreply.github.com>

### CODE REVIEW

File: src/poll/epoll.rs [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+    }
+}
+
+pub enum CallbackMessage {
+    Unregister(i32),
+    HandleExplicitly(i32),
+    Nop,
+}
+
+pub struct Epoll {
```
Comment (dipeshkaphle) [inline src/poll/epoll.rs]: We can remove this no? Since EpollGeneric is handling everything now

File: src/poll/epoll.rs [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+    fds: HashMap<u64, EpollEntryType>,
+    epoll_fd: RawFd,
+}
+
+impl Epoll {
```
Comment (dipeshkaphle) [inline src/poll/epoll.rs]: this too

File: src/poll/epoll.rs [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+            return Ok(None);
+        }
+
+        let event = events.first().unwrap();
+        let fd = event.data();
+        Ok(Some((*event, self.fds.get_mut(&fd).unwrap())))
+    }
+}
+
+impl Drop for Epoll {
```
Comment (dipeshkaphle) [inline src/poll/epoll.rs]: add Drop trait for EpollGeneric and remove this one

**Review by dipeshkaphle — COMMENTED** [review]
Remove the Drop trait implementations for cpp,py,java and simulator as well

**Review by dipeshkaphle — APPROVED** [review]

**Review by Ram-20062003 — APPROVED** [review]

**Pull request merged**